libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -1430,165 +1432,165 @@ float AC_PosControl::crosstrack_error() const
 // todo: this should be based on thrust vector attitude control
-void AC_PosControl::lean_angles_to_accel_xy(float& accel_x_cmss, float& accel_y_cmss) const
+void AC_PosControl::lean_angles_to_accel_ne_cmss(float& accel_n_cmss, float& accel_e_cmss) const
 {
     // rotate our roll, pitch angles into lat/lon frame
     Vector3f att_target_euler = _attitude_control.get_att_target_euler_rad();
     att_target_euler.z = _ahrs.yaw;
-    Vector3f accel_cmss = lean_angles_to_accel(att_target_euler);
+    Vector3f accel_ne_cmss = lean_angles_to_accel(att_target_euler);
 
-    accel_x_cmss = accel_cmss.x;
-    accel_y_cmss = accel_cmss.y;
+    accel_n_cmss = accel_ne_cmss.x;
+    accel_e_cmss = accel_ne_cmss.y;
 }
 
 // calculate_yaw_and_rate_yaw - update the calculated the vehicle yaw and rate of yaw.
 void AC_PosControl::calculate_yaw_and_rate_yaw()
 {
     // Calculate the turn rate
-    float turn_rate = 0.0f;
-    const float vel_desired_xy_len = _vel_desired.xy().length();
-    if (is_positive(vel_desired_xy_len)) {
-        const float accel_forward = (_accel_desired.x * _vel_desired.x + _accel_desired.y * _vel_desired.y) / vel_desired_xy_len;
-        const Vector2f accel_turn = _accel_desired.xy() - _vel_desired.xy() * accel_forward / vel_desired_xy_len;
-        const float accel_turn_xy_len = accel_turn.length();
-        turn_rate = accel_turn_xy_len / vel_desired_xy_len;
-        if ((accel_turn.y * _vel_desired.x - accel_turn.x * _vel_desired.y) < 0.0) {
-            turn_rate = -turn_rate;
+    float turn_rate_rads = 0.0f;
+    const float vel_desired_length_ne_cms = _vel_desired_neu_cms.xy().length();
+    if (is_positive(vel_desired_length_ne_cms)) {
+        const float accel_forward = (_accel_desired_neu_cmss.x * _vel_desired_neu_cms.x + _accel_desired_neu_cmss.y * _vel_desired_neu_cms.y) / vel_desired_length_ne_cms;
+        const Vector2f accel_turn_ne_cmss = _accel_desired_neu_cmss.xy() - _vel_desired_neu_cms.xy() * accel_forward / vel_desired_length_ne_cms;
+        const float accel_turn_length_ne_cmss = accel_turn_ne_cmss.length();
+        turn_rate_rads = accel_turn_length_ne_cmss / vel_desired_length_ne_cms;
+        if ((accel_turn_ne_cmss.y * _vel_desired_neu_cms.x - accel_turn_ne_cmss.x * _vel_desired_neu_cms.y) < 0.0) {
+            turn_rate_rads = -turn_rate_rads;
         }
     }
 
-    // update the target yaw if velocity is greater than 5% _vel_max_xy_cms
-    if (vel_desired_xy_len > _vel_max_xy_cms * 0.05f) {
-        _yaw_target = degrees(_vel_desired.xy().angle()) * 100.0f;
-        _yaw_rate_target = turn_rate * degrees(100.0f);
+    // update the target yaw if velocity is greater than 5% _vel_max_ne_cms
+    if (vel_desired_length_ne_cms > _vel_max_ne_cms * 0.05f) {
+        _yaw_target_cd = degrees(_vel_desired_neu_cms.xy().angle()) * 100.0f;
+        _yaw_rate_target_cds = degrees(turn_rate_rads) * 100.0f;
```

Comment:
This may be the reason the output isn't binary-output-identical.  It's also worse than the existing code as it has to be done at runtime rather than at compile-time